altair/expr/__init__.py

```diff
@@ -110,1321 +114,1561 @@ class expr(_ExprRef, metaclass=_ConstExpressionType):
+
+        .. _d3-format specifier:
+            https://github.com/d3/d3-format/
         """
-        return FunctionExpression("format", args)
+        return FunctionExpression("format", (value, specifier))
 
     @classmethod
-    def monthFormat(cls, *args) -> FunctionExpression:
+    def monthFormat(cls, month: IntoExpression, /) -> Expression:
         """
-        Formats a (zero-based) *month* number as a full month name, according to the current locale.
+        Formats a (zero-based) ``month`` number as a full month name, according to the current locale.
 
-        For example: `monthFormat(0) -> "January"`.
+        For example: ``alt.expr.monthFormat(0) -> "January"``.
         """
-        return FunctionExpression("monthFormat", args)
+        return FunctionExpression("monthFormat", (month,))
 
     @classmethod
-    def monthAbbrevFormat(cls, *args) -> FunctionExpression:
+    def monthAbbrevFormat(cls, month: IntoExpression, /) -> Expression:
         """
-        Formats a (zero-based) *month* number as an abbreviated month name, according to the current locale.
+        Formats a (zero-based) ``month`` number as an abbreviated month name, according to the current locale.
 
-        For example: `monthAbbrevFormat(0) -> "Jan"`.
+        For example: ``alt.expr.monthAbbrevFormat(0) -> "Jan"``.
         """
-        return FunctionExpression("monthAbbrevFormat", args)
+        return FunctionExpression("monthAbbrevFormat", (month,))
 
     @classmethod
-    def timeUnitSpecifier(cls, *args) -> FunctionExpression:
+    def timeUnitSpecifier(
+        cls, units: IntoExpression, specifiers: IntoExpression = None, /
+    ) -> Expression:
         """
-        Returns a time format specifier string for the given time `unit <https://vega.github.io/vega/docs/api/time/#time-units>`__.
-
-        The optional *specifiers* object provides a set of specifier sub-strings for customizing the format;
-        for more, see the `timeUnitSpecifier API documentation <https://vega.github.io/vega/docs/api/time/#timeUnitSpecifier>`__.
+        Returns a time format specifier string for the given time `*units*`_.
 
-        The resulting specifier string can then be used as input to the `timeFormat <https://vega.github.io/vega/docs/expressions/#timeFormat>`__ or
-        `utcFormat <https://vega.github.io/vega/docs/expressions/#utcFormat>`__ functions, or as the *format* parameter of an axis or legend.
+        The optional ``specifiers`` object provides a set of specifier sub-strings for customizing
+        the format; for more, see the `timeUnitSpecifier API documentation`_. The resulting
+        specifier string can then be used as input to the `timeFormat`_ or `utcFormat`_ functions,
+        or as the *format* parameter of an axis or legend. For example: ``alt.expr.timeFormat(date,
+        alt.expr.timeUnitSpecifier('year'))`` or ``alt.expr.timeFormat(date,
+        alt.expr.timeUnitSpecifier(['hours', 'minutes']))``.
```

Comment:
## Inline -> block code idea
An example of how this change would show
```py
def func(*args):
	"""
	...
        The optional ``specifiers`` object provides a set of specifier sub-strings for customizing
        the format; for more, see the `timeUnitSpecifier API documentation`_. The resulting
        specifier string can then be used as input to the `timeFormat`_ or `utcFormat`_ functions,
        or as the *format* parameter of an axis or legend. 
		
		Examples
		--------
		>>> alt.expr.timeFormat(date, alt.expr.timeUnitSpecifier('year'))
		>>> alt.expr.timeFormat(date, alt.expr.timeUnitSpecifier(['hours', 'minutes']))
	"""
	...
```

## Issues
- Some of these uses the pattern `"... -> result"`
	- This can be recreated using the doctest syntax
	- but the actual result won't be the same as what is stated, since they are not evaluated
- `alt.expr.format` uses a different style
	- "(e.g., ``alt.expr.format(value, ',.2f')``"
	- Whereas all the others use (`"For example:..."` | `"Example:..."`)